### PR TITLE
UI: Makes 'Add [Support|Chat] Thread' floating

### PR DESCRIPTION
**Pull request by aethernet**
UI: Makes Add Thread button floating to save screen real estate.

Change-type: patch

### CODE REVIEW

**Comment by joshbwlng** [comment]
@aethernet Could you include some before/after screenshots in your PR description to make it clear what "floating" means? Thanks!

File: apps/ui/lib/components/InfiniteList.tsx [hunk 1]
```diff
@@ -15,7 +15,9 @@ const ScrollArea = styled<any>(Box)`
 	overflow-y: auto;
 	display: flex;
 	flex-direction: column;
+	padding-bottom: 100px;
```
Comment (LucianBuzzo) [inline apps/ui/lib/components/InfiniteList.tsx:18]: The additional bottom padding works fine here - its only an issue when you're at the most recent message

File: apps/ui/lib/lens/list/Interleaved/component.tsx [hunk 2]
```diff
@@ -139,6 +139,20 @@ export const InterleavedList = (props: Props) => {
 			startReached={loadMoreContracts}
 			itemContent={itemContent}
 			overscan={10}
+			components={{
```
Comment (LucianBuzzo) [inline apps/ui/lib/lens/list/Interleaved/component.tsx:142]: What is this Footer component for?

File: apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx [hunk 3]
```diff
@@ -14,7 +14,10 @@ import * as notifications from '../../../services/notifications';
 import { RelationshipContract } from 'autumndb';
 
 const Footer = styled(Flex)`
-	border-top: 1px solid #eee;
+	// border-top: 1px solid #eee;
```
Comment (LucianBuzzo) [inline apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx:17]: You can remove this commented out code

**Comment by aethernet** [comment]
@joshbwlng here's
Cursor_and__52__3a563837-a2ad-4528-9559-5e4______Jellyfish
_52__3a563837-a2ad-4528-9559-5e4______Jellyfish
_52__6a3c84d9-7f5d-407f-a17b-917______Jellyfish

File: apps/ui/lib/lens/list/Interleaved/component.tsx (hunk 2 above)
Comment (aethernet) [reply apps/ui/lib/lens/list/Interleaved/component.tsx:142]: Adds a last "empty" element on the virtuoso list, which have the same purpose as the margin-bottom.

We cannot easily attach the margin-bottom to any of the internal virtuoso divs and doing on the wrapper reduces the scrollview.

Next step for this footer is to add a spinner while loading and a "end of scroll" message (or image)

**Comment by aethernet** [comment]
@joshbwlng before / after 

_52__6a3c84d9-7f5d-407f-a17b-917______Jellyfish-2

_52__3a563837-a2ad-4528-9559-5e4______Jellyfish-2

File: apps/ui/lib/lens/list/Simple/component.tsx [hunk 4]
```diff
@@ -77,6 +77,20 @@ export default function SimpleList(props: Props) {
 			endReached={loadMoreContracts}
 			itemContent={itemContent}
 			overscan={5}
+			components={{
```
Comment (joshbwlng) [inline apps/ui/lib/lens/list/Simple/component.tsx:80]: Looks like you have some duplicate code

**Comment by joshbwlng** [comment]
@aethernet Would these changes not result in a button that would block visibility of text/information under it?

**Comment by aethernet** [comment]
@joshbwlng yes indeed, but it only matter if you cannot scroll to move the test out of under the button.

That's the reason for the margin-bottom (on generic scroll lists) / empty footer (on virtuoso). It ensure you can always scroll the last line out of the button (and gives a little bit of comfort at the end as the last line of the feed is not glue to the side of the window/screen)

**Comment by aethernet** [comment]
_64__user-jmakivic___Jellyfish

File: apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx [hunk 5]
```diff
@@ -14,7 +14,9 @@ import * as notifications from '../../../services/notifications';
 import { RelationshipContract } from 'autumndb';
 
 const Footer = styled(Flex)`
-	border-top: 1px solid #eee;
+	position: absolute;
```
Comment (LucianBuzzo) [inline apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx:17]: This causes problems for lenses that don't infinite scroll, like the table lens. We may need to conditionally fix the button based on the lens being rendered 🤔

**Review by LucianBuzzo — COMMENTED** [review]
I've pushed a commit that cleans up the Footer a little. Overall I like the change, however the flooring button causes layout issues for more static lenses, like the table and we need to resolve that issue before this can be merged.

File: apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx (hunk 5 above)
Comment (LucianBuzzo) [reply apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx:17]: After some though I think the best approach will be to give controls to lenses to declaratively not render the footer, and adjust the other lenses to compensate for a floating button.

**Comment by LucianBuzzo** [comment]
@aethernet I tweaked this a bit more to make it play nicely with the other lenses, and added a similar affect for the segment footer (note to self: we need to unify those components). I also went a step further and switched to using an icon + tooltip for the buttons which makes them much smaller:

image

**Review by LucianBuzzo — APPROVED** [review]

**Comment by aethernet** [comment]
@LucianBuzzo awesome !!

**Pull request merged**